core/src/main/java/com/linecorp/armeria/common/multipart/ContentDispositionBuilder.java

```diff
@@ -0,0 +1,165 @@
+     * be escaped with a backslash, e.g. {@code "foo\"bar.txt"} becomes
+     * {@code "foo\\\"bar.txt"}.
+     */
+    public ContentDispositionBuilder filename(String filename) {
+        return filename(filename, null);
+    }
+
+    /**
+     * Sets the value of the {@code filename*} that will be encoded as defined in the RFC 5987.
+     * Only the US-ASCII, UTF-8 and ISO-8859-1 charsets are supported.
+     *
+     * <p><strong>Note:</strong> Do not use this for a {@code "multipart/form-data"} requests as per
+     * <a link="https://tools.ietf.org/html/rfc7578#section-4.2">RFC 7578, Section 4.2</a>
+     * and also RFC 5987 itself mentions it does not apply to multipart requests.
+     */
+    public ContentDispositionBuilder filename(String filename, Charset charset) {
+        requireNonNull(filename, "filename");
+        checkArgument(!filename.isEmpty(), "filename should not be empty.");
+        this.filename = filename;
+        this.charset = charset;
+        return this;
+    }
+
+    /**
+     * Set the value of the {@code size} parameter.
+     *
+     * @deprecated As per <a href="https://tools.ietf.org/html/rfc6266#appendix-B">RFC 6266, Appendix B</a>,
+     *             to be removed in a future release.
+     */
+    @Deprecated
+    public ContentDispositionBuilder size(Long size) {
+        this.size = size;
+        return this;
+    }
+
+    /**
+     * Set the value of the {@code creation-date} parameter.
+     *
+     * @deprecated As per <a href="https://tools.ietf.org/html/rfc6266#appendix-B">RFC 6266, Appendix B</a>,
+     *             to be removed in a future release.
+     */
+    @Deprecated
+    public ContentDispositionBuilder creationDate(ZonedDateTime creationDate) {
+        this.creationDate = creationDate;
+        return this;
+    }
+
+    /**
+     * Set the value of the {@code modification-date} parameter.
+     *
+     * @deprecated As per <a href="https://tools.ietf.org/html/rfc6266#appendix-B">RFC 6266, Appendix B</a>,
+     *             to be removed in a future release.
+     */
+    @Deprecated
+    public ContentDispositionBuilder modificationDate(ZonedDateTime modificationDate) {
+        this.modificationDate = modificationDate;
+        return this;
+    }
+
+    /**
+     * Set the value of the {@literal read-date} parameter.
+     * @deprecated As per <a href="https://tools.ietf.org/html/rfc6266#appendix-B">RFC 6266, Appendix B</a>,
+     *             to be removed in a future release.
+     */
+    @Deprecated
+    public ContentDispositionBuilder readDate(ZonedDateTime readDate) {
+        this.readDate = readDate;
+        return this;
+    }
+
+    /**
+     * Returns a newly-created {@link ContentDisposition} based on the properties of this builder.
+     */
+    public ContentDisposition build() {
+        return new ContentDisposition(type, name, filename, charset, size,
+                                      creationDate, modificationDate, readDate);
+    }
```

Comment:
Out of curiosity, are all parameters except `type`optional?

Reply:
From the spec, `type` is only the required parameter.
https://www.w3.org/Protocols/rfc2616/rfc2616-sec19.html#sec19.5.1

`name` and `filename` are optional parameters. However, they are used commonly for request headers. https://developer.mozilla.org/ko/docs/Web/HTTP/Headers/Content-Disposition#Syntax
```http
Content-Disposition: form-data
Content-Disposition: form-data; name="fieldName"
Content-Disposition: form-data; name="fieldName"; filename="filename.jpg"
```

I think it would be convenient to offer shortcut factories for them.
I'm worried that those factories only take `String` as parameters.
```java
public static ContentDisposition of(String type) {}

public static ContentDisposition of(String type, String name) {}

public static ContentDisposition of(String type, String name, String filename) {}
```

Reply:
SGTM